jena-rdfconnection/src/main/java/org/apache/jena/rdflink/RDFLinkFuseki.java

```diff
@@ -69,7 +69,7 @@ private static RDFLinkHTTPBuilder setupForFuseki(RDFLinkHTTPBuilder builder) {
             .acceptHeaderSelectQuery(ResultSetLang.RS_Thrift.getHeaderString())
             .acceptHeaderAskQuery(ResultSetLang.RS_JSON.getHeaderString())
             .acceptHeaderQuery(acceptHeaderSPARQL)
-            .parseCheckSPARQL(false)
+            .parseCheckSPARQL(true)
```

Comment:
With `parseCheckSPARQL(false)` there is now one failing test:

`TestRDFLinkFuseki.query_build_02` assumes that `QueryExecBuilder.query(String)` parses the query even though the link was created with `RDFLink.parseCheckSPARQL(false)`.

Perhaps it is the test that should be adapted.

Reply:
+1, sometimes the tests make assumptions about how APIs work that are no longer true when we fix bugs in the API/implementation so the affected tests should be adapted accordingly

Reply:
Thinking about this issue, the code in question is:
```java
QueryExec qExec = link.newQuery().query("SELECT ?X { }")
    .substitution(binding)
    .build();
```

The problem is that the link has checkParse disabled, but substitution requires a query object.
I think it would be best if methods that require a object would attempt to parse a given query string when needed.

The alternative would be having to explicitly write `remoteLink.newQuery().checkParse(true).query("SELECT ?X { }")` which doesn't feel natural.

Reply:
Yeah, so maybe adjust the code and update the Javadoc on the `substitution` method to note that this requires parsing the query which means it can't be used with SPARQL extensions Jena doesn't understand

Reply:
I changed the Query/Update exec builders so that the build() method will attempt to parse any remaining strings if needed.